packages/typespec-client-generator-core/src/rules/require-named-model.rule.ts

```diff
@@ -0,0 +1,34 @@
+import { createRule, Model, paramMessage } from "@typespec/compiler";
+import { createTCGCContext } from "../context.js";
+import { createSdkPackage } from "../package.js";
+
+export const requireNamedModelRule = createRule({
+  name: "require-named-model",
```

Comment:
```suggestion
  name: "no-unnamed-types",
```

I think this name would be better. There is also other cases than models like inline union and in the future we might allow any type to be created as expressions too